phys2bids/physio_obj.py

```diff
@@ -524,9 +524,8 @@ def check_trigger_amount(self, thr=None, num_timepoints_expected=0, tr=0):
             thr = np.mean(trigger)
             flag = 1
         timepoints = trigger > thr
-        num_timepoints_found = len(
-            [is_true for is_true, _ in groupby(timepoints, lambda x: x != 0) if is_true]
-        )
+        # way more efficient alternative
+        num_timepoints_found = np.count_nonzero(np.ediff1d(timepoints.astype(np.int8)) > 0)
```

Comment:
This does look like a better alternative, thank you for that!
We've been refactoring parts of the code recently to make them more numpy-based and perform better, so it's a very well appreciated change.